src/feature/hs/hs_common.c

```diff
@@ -1723,45 +1714,45 @@ hs_get_extend_info_from_lspecs(const smartlist_t *lspecs,
     }
   } SMARTLIST_FOREACH_END(ls);
 
+  fascist_firewall_choose_address_ls(lspecs, FIREWALL_OR_CONNECTION, 0, &ap);
```

Comment:
If direct_conn is not true, we need to choose an IPv4 address, because relays only extend to IPv4 addresses.
See the similar code in:
https://github.com/torproject/tor/blob/8b0920bb6f0d52402026d38dcc8405c0ff263dbb/src/core/or/circuitbuild.c#L2874-L2880
(node_get_prim_orport() always returns the IPv4 address.)